internal/wal_push_handler.go

```diff
@@ -75,9 +141,46 @@ func uploadWALFile(uploader *WalUploader, walFilePath string, preventWalOverwrit
 		return errors.Wrapf(err, "upload: could not open '%s'\n", walFilePath)
 	}
 	err = uploader.UploadWalFile(walFile)
+	if err == nil && (viper.IsSet(UploadWalMetadata) || viper.GetString(UploadWalMetadata) == WalNoMetadataLevel) {
+		err = uploadWALMetadataFile(uploader, walFilePath)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to upload metadata file")
```

Comment:
Should we abort the entire WAL file upload if we failed to upload metadata?

Reply:
WAL metadata may be critical for PITR and that was the reason aborting with error. If agreed, I can change it to log the error and continue the process.